packages/react-storage/src/components/StorageBrowser/views/LocationsView/__tests__/useLocationsView.spec.ts

```diff
@@ -0,0 +1,173 @@
+import { renderHook, act } from '@testing-library/react';
+import { useLocationsView, DEFAULT_LIST_OPTIONS } from '../useLocationsView';
+import { useLocationsData } from '../../../context/actions';
+import { useControl } from '../../../context/control';
+import { usePaginate } from '../../hooks/usePaginate';
+import { LocationAccess } from '../../../context/types';
+import { DataState } from '@aws-amplify/ui-react-core';
+
+jest.mock('../../../context/actions');
+jest.mock('../../../context/control');
+jest.mock('../../hooks/usePaginate');
+
+const mockData: LocationAccess[] = [
+  { scope: 'Location A', type: 'BUCKET', permission: 'READ' },
+  { scope: 'Location B', type: 'PREFIX', permission: 'WRITE' },
+  { scope: 'Location C', type: 'BUCKET', permission: 'READ' },
+  { scope: 'Location D', type: 'PREFIX', permission: 'WRITE' },
+  { scope: 'Location E', type: 'BUCKET', permission: 'READ' },
+];
+const PAGE_SIZE = 3;
+
+function mockUseLocationsData(
+  returnValue: DataState<{ result: LocationAccess[] }>
+) {
+  const handleList = jest.fn();
+  (useLocationsData as jest.Mock).mockReturnValue([returnValue, handleList]);
+  return handleList;
+}
```

Comment:
What does this provide that using a `jest.spyOn` does not? Concerned abt creating test utilities that do not match the expected behavior of the API they are meant to mock

Reply:
Yea, I think replacing it with a spy like the LocationsView test makes sense.